### PR TITLE
ci: do not run north-south conn disrupt for 5.4 kernel

### LINKED ISSUE #40369: [v1.17] CI: Cilium IPsec upgrade - FIB lookup failed
## CI failure

Seems to have started failing recently, always in matrix entry 5:
* https://github.com/cilium/cilium/actions/runs/16053957399/job/45303525286
* https://github.com/cilium/cilium/actions/runs/16060475298/job/45325117752

And similar failures in the E2E Upgrade workflow (`ipsec-5` entry):
* https://github.com/cilium/cilium/actions/runs/16053957092/job/45303510892
* https://github.com/cilium/cilium/actions/runs/16060475163/job/45325114565
* https://github.com/cilium/cilium/actions/runs/16070780875/job/45358111155

Example sysdump: [cilium-sysdump-ipsec-5-final.zip](https://github.com/user-attachments/files/21057962/cilium-sysdump-ipsec-5-final.zip)

```
📋 Test Report [cilium-test-1]
❌ 1/2 tests failed (1/29 actions), 6 tests skipped, 0 scenarios skipped:
Test [no-unexpected-packet-drops]:
  🟥 no-unexpected-packet-drops/no-unexpected-packet-drops:kind-kind/kind-control-plane: Found unexpected packet drops:
{
  "labels": {
    "direction": "EGRESS",
    "reason": "FIB lookup failed"
  },
  "name": "cilium_drop_count_total",
  "value": 2
}
```

### PR DESCRIPTION
**Pull request by ldelossa**
Upstream cilium CLI now runs north-south conn disruptions more liberally.

This was made possible by EoL'ing the 5.4 kernel tests.

However, stable branches will still test 5.4 kernel.

Therefore, we need to ensure north-south conn disrupt tests do not run for stable branches which still place the 5.4 kernel under test.

```release-note
ci: do not run north-south conn disrupt tests for 5.4 kernels
```

### CODE REVIEW

**Review by Artyop — APPROVED** [review]
lgtm

File: .github/actions/conn-disrupt-test-setup/action.yaml [hunk 1]
```diff
@@ -6,22 +6,35 @@ inputs:
...
-          --expected-xfrm-errors "+inbound_no_state"
+          --conn-disrupt-test-restarts-path './cilium-conn-disrupt-restarts' \
+          --conn-disrupt-test-xfrm-errors-path './cilium-conn-disrupt-xfrm-errors' \
+          --expected-xfrm-errors '+inbound_no_state'"
+
+        if [[ "${version}" != "5.4" ]]; then
+          CMD="$CMD --include-conn-disrupt-test-ns-traffic"
+        fi
+
+        eval "$CMD"
```
Comment (nbusseneau) [inline .github/actions/conn-disrupt-test-setup/action.yaml]: Is there any reason we are switching from direct call to `eval` here? I would tend to advise not using `eval` and stick with enriching arguments instead. It should be secure enough here, but I'm thinking of future maintenance where it may be less obvious when changes are made to the workflow how the command is actually called later down the pipeline via `eval`, whereas with the former structure this will not happen.

**Comment by julianwiedmann** [comment]
I'd like to merge this along with the new CLI version once it's released, so we know that it actually works. Let's please park the change until then? :)

File: .github/actions/conn-disrupt-test-setup/action.yaml (hunk 1 above)
Comment (Artyop) [reply .github/actions/conn-disrupt-test-setup/action.yaml]: Building a variable containing `--include-conn-disrupt-test-ns-traffic` and adding it to the end of the command whether it's empty or not could help avoid using `eval`

Comment (ldelossa) [reply .github/actions/conn-disrupt-test-setup/action.yaml]: Yup, I agree. Got a bit `eval` happy there. 

I update to use variable parameter expansion instead.

**Review by nbusseneau — APPROVED** [review]
